### PR TITLE
Update docs and benchmarks to fit with #2182

### PR DESCRIPTION
**Pull request by yeukfu**
Update docs and benchmarks to fit with the new way of constructing the sampler. See https://github.com/rlworkgroup/garage/pull/2182.

### CODE REVIEW

File: docs/user/implement_algo.md [hunk 1]
```diff
@@ -198,11 +198,10 @@ from garage.np import discount_cumsum
 
 class SimpleVPG:
 
-    sampler_cls = LocalSampler
-
-    def __init__(self, env_spec, policy):
+    def __init__(self, env_spec, policy, sampler):
         self.env_spec = env_spec
         self.policy = policy
+        self.sampler = sampler
```
Comment (krzentner) [inline docs/user/implement_algo.md:204]: Can you make all of the fields here have leading `_`? It demonstrates that they're not part of the policy API.

File: docs/user/sampling.md [hunk 2]
```diff
@@ -40,34 +40,76 @@ on a single CPU. `VecWorker` can compute a batch of actions from a policy
...
+In the above example, we firstly construct a worker factory, which will
+construct 4 workers for the sampler. And the max length of episodes collected
+by these workers will be 100. Noted that for policies with TensorFLow
+framework, we need to set `is_tf_worker` to be `True`. Here we don't choose
+a type of worker explicitly, so it will construct `DefaultWorker` by default.
+
+With the worker factory, then we construct a `LocalSampler` with policies and
+environments that will be used in sampling.
+
+### From parameters
```
Comment (krzentner) [inline docs/user/sampling.md:73]: Can you actually swap the order of these two sections? It's now much more common to construct a sampler directly.

**Review by krzentner — APPROVED** [review]
This looks great. I'm really glad we're getting this API change done completely. I have a few small comments on how the docs could be a little better. Please feel free to merge them after you've addressed them (and gotten a review from someone else).

File: docs/user/implement_algo.md (hunk 1 above)
Comment (yeukfu) [reply docs/user/implement_algo.md:204]: But we need to access algo.sampler in the trainer. Would it be appropriate to make this private? And before this change the `sample_cls` field also did't have leading `_`.

**Review by ryanjulian — APPROVED** [review]

File: docs/user/implement_algo.md (hunk 1 above)
Comment (ryanjulian) [reply docs/user/implement_algo.md:204]: It should generally be a read-only property.

**Pull request merged**